native_words.asm

```diff
@@ -8088,14 +8050,10 @@ z_refill:       rts
 xt_repeat:
                 ; Run again first
                 jsr xt_again
-
                 ; Stuff HERE in for the branch address
                 ; to get out of the loop
-                jsr xt_here
-                jsr xt_swap
-                jsr xt_store
-
-z_repeat:       rts
+                jmp xt_then
```

Comment:
You could just move the label and code for `xt_then` right here and save the JMP.  Add a `z_then` label down along with z_repeat.  This essentially overlaps the two words with `xt_repeat` including a `JSR xt_again` first (two entry points, one shared exit point).

Reply:
I ended up merging it with xt_else / xt_endof since the former is probably used more and arguably easier to read the code with else and then nearby.